packages/kbn-alerts-ui-shared/src/rule_type_modal/components/helpers/filter_and_count_rule_types.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { countBy } from 'lodash';
+import {
+  RuleTypeIndexWithDescriptions,
+  RuleTypeCountsByProducer,
+  RuleTypeWithDescription,
+} from '../../types';
+
+export const filterAndCountRuleTypes: (
+  ruleTypeIndex: RuleTypeIndexWithDescriptions,
+  selectedProducer: string | null,
+  searchString: string
+) => [RuleTypeWithDescription[], RuleTypeCountsByProducer] = (
+  ruleTypeIndex,
+  selectedProducer,
+  searchString
+) => {
+  const ruleTypeValues = [...ruleTypeIndex.values()];
+  if (!ruleTypeValues.length) return [[], { total: 0 }];
+
+  const ruleTypesFilteredBySearch = ruleTypeValues.filter((ruleType) => {
+    if (searchString) {
+      return (
+        ruleType.name.toLowerCase().includes(searchString.toLowerCase()) ||
```

Comment:
nitpick: let's save the lowercased string to a variable so we don't have to repeat this